pkg/microservice/aslan/core/environment/service/revision.go

```diff
@@ -129,6 +128,61 @@ func ListProductsRevision(productName, envName string, userID int, superUser boo
 	return prodRevs, nil
 }
 
+func ListPmTemplateProductsRevision(basicFacility string, log *zap.SugaredLogger) ([]*ProductRevision, error) {
+	var (
+		err      error
+		prodRevs = make([]*ProductRevision, 0)
+		products = make([]*commonmodels.Product, 0)
+	)
+
+	temProducts, err := templaterepo.NewProductColl().ListWithOption(&templaterepo.ProductListOpt{BasicFacility: basicFacility})
+	if err != nil {
+		log.Errorf("Collection.TemplateProduct.List error: %s", err)
+		return prodRevs, e.ErrListProducts.AddDesc(err.Error())
+	}
+
+	for _, v := range temProducts {
+		productsT, err := commonrepo.NewProductColl().List(&commonrepo.ProductListOptions{ExcludeStatus: setting.ProductStatusDeleting, Name: v.ProductName})
+		if err != nil {
+			log.Errorf("Collection.Product.List error: %s", err)
+			return prodRevs, e.ErrListProducts.AddDesc(err.Error())
+		}
+		products = append(products, productsT...)
+	}
+
+	// 获取所有服务模板最新模板信息
+	allServiceTmpls, err := commonrepo.NewServiceColl().ListAllRevisions()
+	if err != nil {
+		log.Errorf("ListAllRevisions error: %s", err)
+		return prodRevs, e.ErrListProducts.AddDesc(err.Error())
+	}
+
+	// 获取所有渲染配置最新模板信息
+	allRenders, err := commonrepo.NewRenderSetColl().ListAllRenders()
```

Comment:
do not use these two methods, you can find more accurate query rules to get what you want.